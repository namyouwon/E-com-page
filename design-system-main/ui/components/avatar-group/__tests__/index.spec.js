/* eslint-env jest */
import React from 'react';
import {
  AvatarGroup,
  AvatarGroupContent,
  AvatarGrouped,
  AvatarImage,
  AvatarGroupedIcon,
  AvatarGroupedUserInitials
} from '../';
import createHelpers from '../../../../jest.helpers';
const { matchesMarkup } = createHelpers(__dirname);

it('renders a default avatar group', () =>
  matchesMarkup(
    <AvatarGroup>
      <AvatarGroupContent type="image" />
    </AvatarGroup>
  ));

it('renders a large avatar group with image and icon', () =>
  matchesMarkup(
    <AvatarGroup size="large">
      <AvatarGrouped isPrimary>
        <AvatarImage />
      </AvatarGrouped>
      <AvatarGrouped>
        <AvatarGroupedIcon />
      </AvatarGrouped>
    </AvatarGroup>
  ));

it('renders a small avatar group with icon and initials', () =>
  matchesMarkup(
    <AvatarGroup size="small">
      <AvatarGrouped isPrimary>
        <AvatarGroupedIcon />
      </AvatarGrouped>
      <AvatarGrouped>
        <AvatarGroupedUserInitials />
      </AvatarGrouped>
    </AvatarGroup>
  ));

it('renders an x-small avatar group with initials and icon', () =>
  matchesMarkup(
    <AvatarGroup size="x-small">
      <AvatarGrouped isPrimary>
        <AvatarGroupedUserInitials />
      </AvatarGrouped>
      <AvatarGrouped>
        <AvatarImage />
      </AvatarGrouped>
    </AvatarGroup>
  ));

it('renders an avatar group with inversed user initials', () =>
  matchesMarkup(
    <AvatarGroup>
      <AvatarGroupContent type="user-initials" isInverse />
    </AvatarGroup>
  ));
