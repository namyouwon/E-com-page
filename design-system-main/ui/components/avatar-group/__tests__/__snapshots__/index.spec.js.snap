// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a default avatar group 1`] = `
<span
  className="slds-avatar-group"
>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__primary"
  >
    <img
      alt="Person 2 name"
      src="/assets/images/avatar2.jpg"
      title="Person 2 name"
    />
  </span>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__secondary"
  >
    <img
      alt="Person 2 name"
      src="/assets/images/avatar2.jpg"
      title="Person 2 name"
    />
  </span>
</span>
`;

exports[`renders a large avatar group with image and icon 1`] = `
<span
  className="slds-avatar-group slds-avatar-group_large"
>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__primary"
  >
    <img
      alt="Person 1 name"
      src="/assets/images/avatar1.jpg"
      title="Person 1 name"
    />
  </span>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__secondary"
  >
    <span
      className="slds-icon_container slds-icon-standard-user slds-avatar-grouped__icon"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon"
      >
        <use
          xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#user"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Description of icon when needed
      </span>
    </span>
  </span>
</span>
`;

exports[`renders a small avatar group with icon and initials 1`] = `
<span
  className="slds-avatar-group slds-avatar-group_small"
>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__primary"
  >
    <span
      className="slds-icon_container slds-icon-standard-user slds-avatar-grouped__icon"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon"
      >
        <use
          xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#user"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Description of icon when needed
      </span>
    </span>
  </span>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__secondary"
  >
    <abbr
      className="slds-avatar__initials slds-avatar-grouped__initials"
      title="Person name"
    >
      WW
    </abbr>
  </span>
</span>
`;

exports[`renders an avatar group with inversed user initials 1`] = `
<span
  className="slds-avatar-group"
>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__primary slds-avatar-grouped_inverse"
  >
    <abbr
      className="slds-avatar__initials slds-avatar__initials_inverse slds-avatar-grouped__initials"
      title="Person name"
    >
      WW
    </abbr>
  </span>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__secondary slds-avatar-grouped_inverse"
  >
    <abbr
      className="slds-avatar__initials slds-avatar__initials_inverse slds-avatar-grouped__initials"
      title="Person name"
    >
      WW
    </abbr>
  </span>
</span>
`;

exports[`renders an x-small avatar group with initials and icon 1`] = `
<span
  className="slds-avatar-group slds-avatar-group_x-small"
>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__primary"
  >
    <abbr
      className="slds-avatar__initials slds-avatar-grouped__initials"
      title="Person name"
    >
      WW
    </abbr>
  </span>
  <span
    className="slds-avatar slds-avatar-grouped slds-avatar-grouped__secondary"
  >
    <img
      alt="Person 1 name"
      src="/assets/images/avatar1.jpg"
      title="Person 1 name"
    />
  </span>
</span>
`;
